packages/purpose-platformstate-writer/src/utils.ts

```diff
@@ -97,6 +131,403 @@ export const deletePlatformPurposeEntry = async (
+        : "";
+
+      // Descriptor data from platform-states
+      const isDescriptorDataMissingInTokenTable =
+        platformAgreementEntry &&
+        catalogEntry &&
+        (!entry.GSIPK_eserviceId_descriptorId ||
+          !entry.descriptorAudience ||
+          !entry.descriptorState);
+
+      const descriptorExpressionAttributeValues: Record<
+        string,
+        AttributeValue
+      > = isDescriptorDataMissingInTokenTable
+        ? {
+            ":GSIPK_eserviceId_descriptorId": {
+              S: makeGSIPKEServiceIdDescriptorId({
+                eserviceId: purpose.eserviceId,
+                descriptorId: unsafeBrandId<DescriptorId>(
+                  platformAgreementEntry.agreementDescriptorId
+                ),
+              }),
+            },
+            ":descriptorState": {
+              S: catalogEntry.state,
+            },
+            ":descriptorAudience": {
+              L: catalogEntry.descriptorAudience.map((item) => ({
+                S: item,
+              })),
+            },
+            ":descriptorVoucherLifespan": {
+              N: catalogEntry.descriptorVoucherLifespan.toString(),
+            },
+          }
+        : {};
+      const descriptorUpdateExpression = catalogEntry
+        ? `, GSIPK_eserviceId_descriptorId = :GSIPK_eserviceId_descriptorId, 
+        descriptorState = :descriptorState, 
+        descriptorAudience = :descriptorAudience, 
+        descriptorVoucherLifespan = :descriptorVoucherLifespan`
+        : "";
+
+      const input: UpdateItemInput = {
+        ConditionExpression: "attribute_exists(GSIPK_purposeId)",
+        Key: {
+          PK: {
+            S: tokenEntryPK,
+          },
+        },
+        ExpressionAttributeValues: {
+          ...agreementExpressionAttributeValues,
+          ...descriptorExpressionAttributeValues,
+          ":newState": {
+            S: purposeState,
+          },
+          ":newPurposeVersionId": {
+            S: purposeVersionId,
+          },
+          ":newUpdateAt": {
+            S: new Date().toISOString(),
+          },
+        },
+        UpdateExpression:
+          "SET purposeState = :newState, purposeVersionId = :newPurposeVersionId, updatedAt = :newUpdateAt" +
+          agreementUpdateExpression +
+          descriptorUpdateExpression,
+        TableName: config.tokenGenerationReadModelTableNameTokenGeneration,
+        ReturnValues: "NONE",
+      };
+      const command = new UpdateItemCommand(input);
+      await dynamoDBClient.send(command);
+    }
+
+    if (result.lastEvaluatedKey) {
+      await runPaginatedUpdateQuery(
+        dynamoDBClient,
+        purpose,
+        purposeState,
+        purposeVersionId,
+        result.lastEvaluatedKey
+      );
+    }
+  };
+
+  await runPaginatedUpdateQuery(
+    dynamoDBClient,
+    purpose,
+    purposeState,
+    purposeVersionId
+  );
+};
+
+export const updatePurposeDataInTokenGenerationStatesTable = async ({
```

Comment:
by just reading the name it is not clear the difference with `updatePurposeEntriesInTokenGenerationStatesTable`